### PR TITLE
feat: refactor influxdb connector to to avoid resources leaking

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes [EMQX-9935](https://emqx.atlassian.net/browse/EMQX-9935)

### <samp>🤖 Generated by Copilot at 85871a7</samp>

Refactor the `emqx_bridge_influxdb_connector` module to use the `emqx_resource` module and update the version and changelog. This improves the reliability and consistency of the influxdb bridge connector.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: changes/ee/feat-10923.en.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactored influxdb bridge connector to avoid resource leaks during crashes during creation.
```
Comment (zmstone) [inline changes/ee/feat-10923.en.md:1]:
```suggestion
Refactored influxdb bridge connector to avoid resource leaks during crashes at creation.
```

**Review by zmstone — APPROVED** [review]

File: apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl [hunk 2]
```diff
@@ -53,10 +56,20 @@
...
+    %% ecpool for influxdb-udp
+    %% See: influxdb:start_client/1
     start_client(InstId, Config).
 
-on_stop(_InstId, #{client := Client}) ->
-    influxdb:stop_client(Client).
+on_stop(InstId, _State) ->
+    case emqx_resource:get_allocated_resources(InstId) of
+        #{?influx_client := Client} ->
+            influxdb:stop_client(Client);
```
Comment (zmstone) [inline apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl:69]: will this return 'ok' if the client is already stopped ?

**Comment by yanzhiemq** [comment]
_

**Comment by yanzhiemq** [comment]
_

**Pull request merged**